docs/deploying.md

```diff
@@ -2,155 +2,135 @@
+## Running a domain registry
 
-You basically have three different options to comply with that security requirement here.
+While running on `localhost` has its uses, most people want their registry to be more widely available. To do so, the Docker engine requires you to secure it using TLS, which is conceptually very similar to configuring your web server with SSL.
 
-### 1. buy a SSL certificate for your domain
+### Get a certificate
 
-This is the (highly) recommended solution.
+Assuming that you own the domain `myregistrydomain.com`, and that its DNS record points to the host where you are running your registry, you first need to get a certificate from a CA.
 
-You can buy a certificate for as cheap as 10$ a year (some registrars even offer certificates for free), and this will save you a lot of trouble.
+Move and/or rename your crt file to: `certs/domain.crt` - and your key file to: `certs/domain.key`.
 
-Assuming you now have a `domain.crt` and `domain.key` inside a directory named `certs`:
+Make sure you stopped your registry from the previous steps, then start your registry again with TLS enabled:
 
-```
-# Stop your registry
-docker stop registry && docker rm registry
-
-# Start your registry with TLS enabled
-docker run -d -p 5000:5000 \
-	-v `pwd`/certs:/certs \
-	-e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt \
-	-e REGISTRY_HTTP_TLS_KEY=/certs/domain.key \
-	--restart=always --name registry \
-	registry:2
-```
+	docker run -d -p 5000:5000 --restart=always --name registry \
+	  -v `pwd`/certs:/certs \
+	  -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt \
+	  -e REGISTRY_HTTP_TLS_KEY=/certs/domain.key \
+	  registry:2
 
-A certificate issuer may supply you with an *intermediate* certificate. In this case, you must combine your certificate with the intermediate's to form a *certificate bundle*. You can do this using the `cat` command: 
+You should now be able to access your registry from another docker host:
 
-```
-$ cat server.crt intermediate-certificates.pem > server.with-intermediate.crt
-```
+    docker pull ubuntu
+    docker tag ubuntu myregistrydomain.com:5000/ubuntu
+    docker push myregistrydomain.com:5000/ubuntu
+    docker pull myregistrydomain.com:5000/ubuntu
 
-You then configure the registry to use your certificate bundle by providing the `REGISTRY_HTTP_TLS_CERTIFICATE` environment variable.
+### Alternatives
 
-**Pros:**
+While rarely advisable, you may want to use self-signed certificates instead, or use your registry in an insecure fashion. You will find instructions [here](insecure.md).
 
- - best solution
- - work without further ado (assuming you bought your certificate from a CA that is trusted by your operating system)
+## Restricting access
 
-**Cons:**
+While running an unrestricted registry is certainly ok for development, secured local networks, or test setups, you should probably implement access restriction if you plan on making it available to a wider audience or through public internet.
```

Comment:
I'd say "Except for registries running on secure local networks, registries should always implement access restrictions." We don't want to give people the impression that if they are doing development or testing, they don't have to worry about security.